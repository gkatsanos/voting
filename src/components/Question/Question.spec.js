import Vue from 'vue';
import { mount } from 'vue-test-utils';
import Question from './Question';

Vue.prototype.$vuetify = {
  load: fn => fn(),
  breakpoint: {},
};

describe('Question.spec.js', () => {
  let cmp;
  let template;

  beforeEach(() => {
    cmp = mount(Question, {
      propsData: { // Replace data value with this fake data
        question: {
          url: '/questions/6673',
          published_at: '2017-10-25T18:20:59.459026+00:00',
          choices: [
            {
              votes: 0,
              url: '/questions/6673/choices/24969',
              choice: 'Objective-C',
            },
            {
              votes: 0,
              url: '/questions/6673/choices/24968',
              choice: 'Python',
            },
            {
              votes: 0,
              url: '/questions/6673/choices/24970',
              choice: 'Ruby',
            },
            {
              votes: 0,
              url: '/questions/6673/choices/24967',
              choice: 'Swift',
            },
          ],
          question: 'Favourite programming language?',
        },
        id: '/questions/6673',
      },
    });
    template = cmp.html();
  });

  it('has the expected html structure', () => {
    expect(template).toMatchSnapshot();
  });

  it('has a link with its UID', () => {
    expect(cmp.vm.$el.querySelector('.card-link').getAttribute('to')).toBe('/questions/6673');
  });
});
