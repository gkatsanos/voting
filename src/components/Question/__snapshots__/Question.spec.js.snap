// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question.spec.js has the expected html structure 1`] = `
"<v-flex d-flex=\\"\\" xs12=\\"\\" md6=\\"\\" xl4=\\"\\"><router-link to=\\"/questions/6673\\" tag=\\"div\\" class=\\"card-link card\\"><v-card hover=\\"\\" class=\\"blue-grey darken-1\\"><v-toolbar dark=\\"\\"><v-toolbar-title class=\\"ml-3\\">
          Favourite programming language?
        </v-toolbar-title> <v-spacer></v-spacer> <!----></v-toolbar> <v-card-text primary-title=\\"\\"><div><div class=\\"headline\\"></div> <div>
            created 2017-10-25T18:20:59.459026+00:00
          </div> <div>
            4 choices
          </div> <!----></div></v-card-text></v-card></router-link></v-flex>"
`;
